app/controllers/reports_controller.rb

```diff
@@ -20,6 +20,7 @@ def history
       else
         @records = GameHistoryDao::get_per_game_histories(primary_player_id)
       end
+      @records = @records.sort_by{|k,v| k}.reverse
```

Comment:
It means that the number of objects in the collection has no upper bound and that it could be a very large number.